test/collateral-fund.js

```diff
@@ -0,0 +1,221 @@
+            amountFundedByOwner = await collateralFundContract.getUserCollateralAmount(owner.address);
+            expect(amountFundedByOwner).to.equal(100);
+
+            amountFundedByUser = await collateralFundContract.getUserCollateralAmount(user.address);
+            expect(amountFundedByUser).to.equal(0);
+        });
+    });
+
+    describe("GetFunders", () => {
+        it("Should add the owner to the funders list", async () => {
+            // Check that the owner and the user are not funders.
+            let funders = await collateralFundContract.getFunders();
+            expect(funders.includes(owner.address)).to.be.false;
+            expect(funders.includes(user.address)).to.be.false;
+
+            // Approve the collateral fund to transfer 100 USDC from the owner's wallet.
+            let txn = await usdcTokenContract.approve(collateralFundContract.address, 100);
+            await txn.wait();
+
+            // Send 100 USDC collateral from the owner's wallet to the fund.
+            txn = await collateralFundContract.deposit(100);
+            await txn.wait();
+
+            // Check that owner is added to the funders list.
+            // Also check that the user is still not a funder.
+            funders = await collateralFundContract.getFunders();
+            expect(funders.includes(owner.address)).to.be.true;
+            expect(funders.includes(user.address)).to.be.false;
+        });
+    })
+
+    /**************************************** Setters ****************************************/
+
+    describe("SetUserCollateralAmount", () => {
+		it("Should update the user collateral amount", async () => {
+            // Check that the user collateral amount is empty.
+            let amountDeposited = await collateralFundContract.getUserCollateralAmount(user.address);
+            expect(amountDeposited).to.equal(0);
+
+			// Update the collateral amount.
+			const txn = await collateralFundContract.connect(hub).setUserCollateralAmount(user.address, 100);
+			await txn.wait();
+
+            // Check that the user collateral amount has been updated.
+            amountDeposited = await collateralFundContract.getUserCollateralAmount(user.address);
+            expect(amountDeposited).to.equal(100);
+
+			// It should fail when updating with a null address.
+			await expect(collateralFundContract.connect(hub).setUserCollateralAmount(ethers.constants.AddressZero, 100))
+				.to.be.revertedWith("The address parameter cannot be null");
+
+			// It should fail when updating the collateral amount as any user.
+			await expect(collateralFundContract.setUserCollateralAmount(user.address, 100))
+                .to.be.revertedWith("Only the hub can call this method");
+		});
+	});
+
+    describe("SetUsdcAddress", () => {
+		it("Should update the usdc address", async () => {
+			// Update the USDC address.
+			const txn = await collateralFundContract.setUsdcAddress(temp.address);
```

Comment:
why is .connect(hub) missing?

Reply:
Because only the owner of the contract can update the USDC address.